src/FunctionConfig.ts

```diff
@@ -55,31 +57,33 @@ export class FunctionConfig {
                     errMessage = localize('expectedBindings', 'Expected "bindings" element of type "Array".');
                 } else {
                     this.functionJson = <IFunctionJson>data;
 
                     const inBinding: IFunctionBinding | undefined = this.functionJson.bindings.find((b: IFunctionBinding) => b.direction === BindingDirection.in);
-                    if (inBinding === undefined) {
+                    if (inBinding === undefined && this.functionJson.bindings.length > 0) {
                         // The generated 'function.json' file for C# class libraries doesn't have direction information (by design), so just use the first
-                        this.inBinding = this.functionJson.bindings[0];
+                        this._inBinding = this.functionJson.bindings[0];
                     } else {
-                        this.inBinding = inBinding;
+                        this._inBinding = inBinding;
                     }
 
-                    if (!this.inBinding.type) {
-                        errMessage = localize('inBindingTypeError', 'The binding with direction "in" must have a type.');
-                    } else {
-                        this.inBindingType = this.inBinding.type;
-                        if (this.inBinding.type.toLowerCase() === 'httptrigger') {
-                            this.isHttpTrigger = true;
-                            if (this.inBinding.authLevel) {
-                                const authLevel: HttpAuthLevel | undefined = <HttpAuthLevel>HttpAuthLevel[this.inBinding.authLevel.toLowerCase()];
-                                if (authLevel === undefined) {
-                                    errMessage = localize('unrecognizedAuthLevel', 'Unrecognized auth level "{0}".', this.inBinding.authLevel);
-                                } else {
-                                    this.authLevel = authLevel;
+                    if (this._inBinding) {
+                        if (!this._inBinding.type) {
+                            errMessage = localize('inBindingTypeError', 'The binding with direction "in" must have a type.');
+                        } else {
+                            this._inBindingType = this._inBinding.type;
+                            if (this._inBinding.type.toLowerCase() === 'httptrigger') {
+                                this.isHttpTrigger = true;
+                                if (this._inBinding.authLevel) {
+                                    const authLevel: HttpAuthLevel | undefined = <HttpAuthLevel>HttpAuthLevel[this._inBinding.authLevel.toLowerCase()];
+                                    if (authLevel === undefined) {
+                                        errMessage = localize('unrecognizedAuthLevel', 'Unrecognized auth level "{0}".', this._inBinding.authLevel);
+                                    } else {
+                                        this.authLevel = authLevel;
+                                    }
```

Comment:
It feels like `authLevel` has a union type just for the error handling, which can be handled by checking if the key `this._inBinding.authLevel.toLowerCase()` is in HttpAuthLevel. 
This is just personal preference, feel free to ignore. :)